Devscord.DiscordFramework/Framework/Commands/Services/BotCommandsMatchingService.cs

```diff
@@ -53,19 +62,19 @@ private bool IsMatchedPropertyType(string value, BotCommandPropertyType type)
             {
                 return false;
             }
-            if (type == BotCommandPropertyType.Time && !_exTime.IsMatch(value))
+            if (type == BotCommandPropertyType.Time && !this._exTime.IsMatch(value))
             {
                 return false;
             }
-            if (type == BotCommandPropertyType.UserMention || type == BotCommandPropertyType.ChannelMention && !_exMention.IsMatch(value))
+            if (type == BotCommandPropertyType.UserMention || type == BotCommandPropertyType.ChannelMention && !this._exMention.IsMatch(value))
             {
                 return false;
             }
             if (type == BotCommandPropertyType.SingleWord && value.Contains(' '))
             {
                 return false;
             }
-            if (type == BotCommandPropertyType.Bool && !bool.TryParse(value, out _))
+            if (type == BotCommandPropertyType.Bool && value != null && !bool.TryParse(value, out _))
```

Comment:
czemu wymuszasz żeby było value?
value powinno być null zawsze 

bool może być sprawdzany przecież wcześniej
zauważ że nie wszystko jest sprawdzane w tym miejscu, np Text nie jest

Reply:
no właśnie, value boola będzie prawie zawsze nullem (tylko nie kiedy ktoś wpisze -json true) dlatego jeśli jest null to już nie będzie sprawdzać czy się parsuje null na bool, tylko pójdzie dalej (nie zwróci false)
bez tego porównania trzeba by zawsze wpisywać -help -json true bo inaczej nie dało by się parsować na boola (!bool.TryParse(value, out _))

Reply:
ja bym zrobił to na wcześniejszym etapie
żeby był jakiś if sprawdzający czy jest nullem
i sam fakt że uda sie znaleźć coś o pasującym name
będzie ustawiać true 

jeśli w boolu nie ustawisz true ręcznie
to dostaniesz false
czyli to czego oczekujemy

jest opcjonalne zmienienie na true za pomocą przykładowo `-json`
ale jak ktoś nie zmieni na true, to dostaniemy false